### DISCUSSION TITLE
Waydroid  installation for raspberry pi 4 64bit arm64?

### DISCUSSION THREAD

**Discussion by nolansingroy** [waydroid/waydroid#896] [General]
Does anyone know if these works on the pi 4 running 64 bit os?

**Comment by hxw** [comment]
It did not work for me. I had to `apt install linux-modules-extra-raspi` to get the binder kernel module since it would fail to initialise after initial install.. This is on a 8GB Pi4 running the latest Ubuntu 22.04.2.

I think the problem is the binder because, although the container was running; launching an apk just timed out waiting for the binder manager, and there was never any display. I could attach to the container and that shell see that all the processes like audio and graphics were running. So I think it may be close, perhaps the binder kernel module needs to be rebuilt, or the manager program is missing/incompatible. That is as far as I got with it.

I did give the anbox package a quick try since that is just an `apt install anbox` but that says Pi4 is not supported. I was doing this install to try to find that binder module since I suspected that the install would build the module, however it turned out to be in the modules extra.

**Comment by theldap** [comment]
Install linux-modules-extra-raspi to provide required kernel features.

Then edit /boot/firmware/cmdline.txt and add psi=1 to the list of kernel parameters and reboot.
This will ensure Waydroid completes its launch.

**Reply by 9acca9** [reply]
@theldap 
hi. It will work to run apps that use:
Hardware accelerated graphics (V3D, OpenGL & Vulkan)
Hardware video decoding & encoding (H.265 decoding, H.264 decoding & encoding)

Also, is possible to run a TV version? or just mobile?
Thanks!

**Reply by boomzo79** [reply]
> Install linux-modules-extra-raspi to provide required kernel features.
> 
> Then edit /boot/firmware/cmdline.txt and add psi=1 to the list of kernel parameters and reboot. This will ensure Waydroid completes its launch.

Hi, I'm from Argentina. I apologize for my English. I wanted to know if there is another way because I did what you recommended and it didn't work well for me.
